tools/clang/lib/CodeGen/CGHLSLMS.cpp

```diff
@@ -5498,9 +5498,14 @@ void CGMSHLSLRuntime::EmitHLSLOutParamConversionInit(
     const Expr *Arg = E->getArg(i+ArgsToSkip);
     QualType ParamTy = Param->getType().getNonReferenceType();
     bool isObject = dxilutil::IsHLSLObjectType(CGF.ConvertTypeForMem(ParamTy));
+    bool isArray = ParamTy->isArrayType();
+    // Check for array of matrix
+    QualType ParamElTy = ParamTy;
+    while (ParamElTy->isArrayType())
+      ParamElTy = ParamElTy->getAsArrayTypeUnsafe()->getElementType();
+    bool isMatrix = hlsl::IsHLSLVecMatType(ParamElTy);
```

Comment:
isMatrix will be true for vector, is it expected?

Reply:
No, copied from expression below, but didn't change to IsHLSLMatType().  Good catch!